lib/hyperswitch.js

```diff
@@ -77,8 +77,9 @@ function HyperSwitch(options, req, parOptions) {
         this.config = options.conf;
         this.config.user_agent = this.config.user_agent || 'HyperSwitch';
         this._rootReq = null;
-        this._forwardedHeaders = null;
-        this._authService = null;
+        this._requestFilters = [];
+        this._subRequestFilters = [];
+        this.ctx = null;
```

Comment:
Perhaps make this parOptions.ctx || null, so that we can explicitly override the ctx when we create a per-request child? I know that the version here will do the right thing as well, but the transition from null to {} seems a bit more subtle that we'd want.